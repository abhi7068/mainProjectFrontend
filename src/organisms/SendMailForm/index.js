import React, { Component } from 'react';
import { Button, Card, TextField } from '@material-ui/core';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import { withStyles } from '@material-ui/styles';
import { withRouter } from 'react-router-dom';
import { sendGift } from '../../services/giftService';
import SnackBar from '../../atoms/SnackBar';
import PropTypes from 'prop-types';

import { Formik, Field, Form, ErrorMessage } from 'formik';
import * as Yup from 'yup';
const styles = {
  root: {
    width: '300px',
    height: '300px',
    margin: '10% auto',
  },
  label: {
    textTransform: 'capitalize',
  },
  display: {
    margin: '5% ',
  },
};

class Forms extends Component {
  constructor(props) {
    super();
    this.state = {
      open: false,
      disabled: false,
    };
  }

  render() {
    return (
      <Formik
        initialValues={{
          email: '',
        }}
        validationSchema={Yup.object().shape({
          email: Yup.string('Enter your email')
            .email('Enter a valid email')
            .required('Email is required')
            .max(35, 'Email must contain no more then 35 characters')
            .matches(
              /^[A-Za-z].*$/,
              'Email should not start with number orspecial character'
            ),
        })}
        onSubmit={(values) => {
          const {
            match: {
              params: { balance },
            },
          } = this.props;

          const details = JSON.parse(
            localStorage.getItem('storeResponseDetails')
          );
          sendGift(details.id, values.email, balance).then((res) => {
            this.setState({ open: res.data.success, disabled: true });
          });
        }}
      >
        {() => {
          return (
            <div>
              {this.state.open && (
                <SnackBar path="/" name="Successfully sent" />
              )}
              <Card
                classes={{
                  root: this.props.classes.root,
                }}
              >
                <CardHeader title="GIFT CARD" />
                <Form>
                  <Field
                    as={TextField}
                    name="email"
                    label="Email"
                    variant="outlined"
                  />

                  <ErrorMessage
                    name="email"
                    component="div"
                    style={{ color: 'red' }}
                  />
                  <CardContent>
                    <Button
                      data-testId="button"
                      variant="contained"
                      color="primary"
                      type="submit"
                      disabled={this.state.disabled}
                    >
                      Send
                    </Button>
                  </CardContent>
                </Form>
              </Card>
            </div>
          );
        }}
      </Formik>
    );
  }
}
Forms.propTypes = {
  classes: PropTypes.object,
  root: PropTypes.object,
  display: PropTypes.object,
  history: PropTypes.object,
  match: PropTypes.object,
};
export default withRouter(withStyles(styles)(Forms));
