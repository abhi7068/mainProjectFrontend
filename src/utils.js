import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
