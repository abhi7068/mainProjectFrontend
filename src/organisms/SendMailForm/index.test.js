import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { fireEvent } from '@testing-library/react';
import SendMailForm from './index';
import { render } from '../../utils';

it('it renders', () => {
  render(<SendMailForm />);
});
it('should take a snapshot', () => {
  const { asFragment } = render(<SendMailForm />);
  expect(asFragment(render(<SendMailForm />))).toMatchSnapshot();
});

it('it renders with correct text', () => {
  const { getByTestId } = render(<SendMailForm />);
  expect(getByTestId('button')).toHaveTextContent('Send');
});
it('renders with expected length"', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <SendMailForm />
    </BrowserRouter>,
    div
  );
  expect(div.querySelectorAll('input')).toHaveLength(1);
});
it('renders with expected length"', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <SendMailForm />
    </BrowserRouter>,
    div
  );
  expect(div.querySelector('input')).toHaveAttribute('name', 'email');
});
it('to be true', () => {
  const { getByTestId } = render(<SendMailForm />);
  expect(fireEvent.click(getByTestId('button'))).toBe(true);
});
