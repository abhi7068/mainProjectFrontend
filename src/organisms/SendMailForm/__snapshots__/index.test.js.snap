// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root Forms-root-174 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            GIFT CARD
          </span>
        </div>
      </div>
      <form
        action="#"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            Email
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              name="email"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-300 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-302"
              >
                <span>
                  Email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiCardContent-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-testid="button"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Send
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
